provider/pkg/provider/provider.go

```diff
@@ -453,6 +459,35 @@ func (p *cfnProvider) Invoke(ctx context.Context,
 	return &pulumirpc.InvokeResponse{Return: res}, nil
 }
 
+func (p *cfnProvider) getInvokeFunc(ctx context.Context, tok string) (invokeFunc, bool) {
+	cf, ok := p.resourceMap.Functions[tok]
+	if !ok {
+		return nil, false
+	}
+	return func(p *cfnProvider, ctx context.Context, inputs resource.PropertyMap) (resource.PropertyMap, error) {
+		idParts := make([]string, len(cf.Identifiers))
+		for i, v := range cf.Identifiers {
+			pv, ok := inputs[resource.PropertyKey(v)]
+			if !ok {
+				return nil, errors.Errorf("missing identifier property '%s'", v)
+			}
+			if !pv.IsString() {
+				return nil, errors.Errorf("identifier property '%s', expected type string, found %s", v, pv.TypeString())
+			}
+			idParts[i] = pv.StringValue()
+		}
+		identifier := strings.Join(idParts, "|")
```

Comment:
Yuck - I didn't realize this is how cctl handles composite identifiers...

Reply:
Yeah, kinda nasty, but 🤷